src/test/java/com/catcher/core/service/ScheduleServiceTest.java

```diff
@@ -299,6 +303,98 @@ void get_none_user_items() {
         assertEquals(0, result.getItems().size());
     }
 
+    @DisplayName("SUCCESS : 정상 참여 처리")
+    @Test
+    void participate() {
+        // given
+        User user = createUser(createRandomUUID(), createRandomUUID(), createRandomUUID(), createRandomUUID());
+        userRepository.save(user);
+        Schedule schedule = createSchedule(user, location, ScheduleStatus.NORMAL);
+        scheduleRepository.save(schedule);
+
+        // when
+        scheduleService.participateSchedule(user, schedule.getId());
+        flushAndClearPersistence();
+
+        // then
+        assertThat(scheduleParticipantRepository.findByUserAndScheduleId(user.getId(), schedule.getId())).isPresent();
+    }
+
+    @DisplayName("SUCCESS : 정상 참여 처리")
+    @Test
+    void participate_when_already_cancel() {
+        // given
+        User user = createUser(createRandomUUID(), createRandomUUID(), createRandomUUID(), createRandomUUID());
+        userRepository.save(user);
+        Schedule schedule = createSchedule(user, location, ScheduleStatus.NORMAL);
+        scheduleRepository.save(schedule);
+        ScheduleParticipant scheduleParticipant = createScheduleParticipant(schedule, user, ParticipantStatus.CANCEL);
+        scheduleParticipantRepository.save(scheduleParticipant);
+
+        // when
+        scheduleService.participateSchedule(user, schedule.getId());
+        flushAndClearPersistence();
+
+        // then
+        assertThat(scheduleParticipantRepository.findByUserAndScheduleId(user.getId(), schedule.getId())).isPresent();
```

Comment:
```suggestion
        var scheduleParticipant = scheduleParticipantRepository.findByUserAndScheduleId(user.getId(), schedule.getId());
        assertThat(scheduleParticipant)).isPresent();
        assertThat(scheduleParticipant.getPartipantStatus()).isEqual(CANCEL);
```
이 부분과 바로 위에 코드도 이와 같이 status 가 정상 반영되었는지를 확인하는 코드가 필요할거 같습니다.

Reply:
수정했습니다!